cdc/sorter/unified/merger.go

```diff
@@ -404,89 +405,83 @@ func runMerger(ctx context.Context, numSorters int, in <-chan *flushTask, out ch
 		return nil
 	}
 
-	resolvedTsNotifierChan := make(chan struct{}, 1)
-	errg, ctx := errgroup.WithContext(ctx)
+	resolvedTsTicker := time.NewTicker(time.Second * 1)
 
-	errg.Go(func() error {
-		for {
-			var task *flushTask
-			select {
-			case <-ctx.Done():
-				return ctx.Err()
-			case task = <-in:
-			}
+	defer resolvedTsTicker.Stop()
 
-			if task == nil {
-				tableID, tableName := util.TableIDFromCtx(ctx)
-				log.Debug("Merger input channel closed, exiting",
-					zap.Int64("tableID", tableID),
-					zap.String("tableName", tableName))
-				return nil
+	var lastResolvedTs uint64
+	resolvedTsTickFunc := func() error {
+		curResolvedTs := atomic.LoadUint64(&m.minResolvedTs)
+		if curResolvedTs > lastResolvedTs {
+			err := onMinResolvedTsUpdate(curResolvedTs)
+			if err != nil {
+				return errors.Trace(err)
 			}
+		} else if curResolvedTs < lastResolvedTs {
+			log.Panic("resolved-ts regressed in sorter",
+				zap.Uint64("curResolvedTs", curResolvedTs),
+				zap.Uint64("lastResolvedTs", lastResolvedTs))
+		}
+		return nil
+	}
 
-			if !task.isEmpty {
-				pendingSet.Store(task, nil)
-			} // otherwise it is an empty flush
-
-			if lastResolvedTs[task.heapSorterID] < task.maxResolvedTs {
-				lastResolvedTs[task.heapSorterID] = task.maxResolvedTs
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-resolvedTsTicker.C:
+			if err := resolvedTsTickFunc(); err != nil {
+				return err
 			}
-
-			minTemp := uint64(math.MaxUint64)
-			for _, ts := range lastResolvedTs {
-				if minTemp > ts {
-					minTemp = ts
-				}
+		case <-m.resolvedTsNotifierChan:
+			if err := resolvedTsTickFunc(); err != nil {
+				return err
 			}
+		}
+	}
+}
 
-			if minTemp > minResolvedTs {
-				atomic.StoreUint64(&minResolvedTs, minTemp)
-				select {
-				case resolvedTsNotifierChan <- struct{}{}:
-				default:
-				}
-			}
+func (m *sorterMerger) runFlushTask(ctx context.Context, numSorters int, in <-chan *flushTask) error {
```

Comment:
Logically `numSorters` is a property of the `sorterMerger`. Can you put `numSorters` inside `sorterMerger`?

Reply:
done